kolena/detection/_internal/test_run.py

```diff
@@ -75,6 +77,7 @@ class BaseTestRun(ABC, Frozen, WithTelemetry):
     _LoadTestImagesDataFrameClass: Type[DFType] = DFType
 
     @validate_arguments(config=ValidatorConfig)
+    @with_event(event_name=EventAPI.Event.EXECUTE_TEST_RUN)
```

Comment:
should this be decorating `run()` method instead?

Reply:
I think you mean the `test()` method for these use cases? That's a good idea as we will be measuring durations in the outer function instead of this init. Will update shortly